piff/gsobject_model.py

```diff
@@ -0,0 +1,341 @@
+
+        center = image.wcs.toWorld(mom.moments_centroid) - image.wcs.toWorld(image_pos)
+        flag = mom.moments_status
+
+        return flux, center, size, shape, flag
+
+    def moment_fit(self, star, logger=None):
+        """Estimate transformations needed to bring self.gsobj towards given star."""
+        flux, centroid, size, shape, flag = self.hsm_moments(star)
+        if flag:
+            raise RuntimeError("Failed measuring HSM moments of star.")
+
+        dflux = flux/self._fiducial_flux
+        dcentroid = centroid - self._fiducial_centroid
+        dsize = size/self._fiducial_size
+        dshape = shape - self._fiducial_shape
+
+        return dflux, dcentroid.x, dcentroid.y, dsize, dshape.g1, dshape.g2, flag
+
+    def getProfile(self, params):
+        """Get a version of the model as a GalSim GSObject
+
+        :param params:      A numpy array with either  [ size, g1, g2 ]
+                            or  [ cenu, cenv, size, g1, g2 ]
+                            depending on if the center of the model is being forced to (0.0, 0.0)
+                            or not.
+
+        :returns: a galsim.GSObject instance
+        """
+        if self._force_model_center:
+            size, g1, g2 = params
+            cenu, cenv = (0.0, 0.0)
+        else:
+            cenu, cenv, size, g1, g2 = params
+        return self.gsobj.dilate(size).shear(g1=g1, g2=g2).shift(cenu, cenv)
+
+    def draw(self, star):
+        """Draw the model on the given image.
+
+        :param star:    A Star instance with the fitted parameters to use for drawing and a
+                        data field that acts as a template image for the drawn model.
+
+        :returns: a new Star instance with the data field having an image of the drawn model.
+        """
+        prof = self.getProfile(star.fit.params).shift(star.fit.center) * star.fit.flux
+        image = star.image.copy()
+        prof.drawImage(image, method='no_pixel', offset=(star.image_pos-image.trueCenter()))
+        data = StarData(image, star.image_pos, star.weight, star.data.pointing)
+        return Star(data, star.fit)
+
+    def _lmfit_resid(self, lmparams, star):
+        """Residual function to use with lmfit.  Essentially `chi` from `chisq`, but not summed
+        over pixels yet.
+
+        :param lmparams:  An lmfit.Parameters() instance.  The model.
+        :param star:    A Star instance.  The data.
+
+        :returns: `chi` as a flattened numpy array.
+        """
+        image, weight, image_pos = star.data.getImage()
+        flux, cenu, cenv, size, g1, g2 = lmparams.valuesdict().values()
+        # Fit cenu and cenv regardless of force_model_center.  The difference is whether the fit
+        # value is recorded (force_model_center=False) or discarded (force_model_center=True).
+        prof = self.gsobj.dilate(size).shear(g1=g1, g2=g2).shift(cenu, cenv) * flux
+        model_image = image.copy()
+        prof.drawImage(model_image, method='no_pixel',
```

Comment:
Sure.  Most of the tests only make sense for `include_pixel=False`, but I think that's okay.  To make sure we're covered, I added a new test for when `include_pixel=True`.